src/Microsoft.Identity.Client/Logger.cs

```diff
@@ -0,0 +1,169 @@
+        }
+
+        public Logger()
+        {
+        }
+
+        internal Guid CorrelationId { get; set; }
+
+        /// <summary>
+        /// The default log level is set to info.
+        /// </summary>
+        internal LogLevel ApplicationLogLevel { get; set; } = LogLevel.Info;
+
+        #region LogMessages
+        /// <summary>
+        /// Method for error logging
+        /// </summary>
+        public void Error(string message)
+        {
+            LogMessage(message, LogLevel.Error, false);
+        }
+
+        /// <summary>
+        /// Method for error logging of Pii 
+        /// </summary>
+        public void ErrorPii(string message)
+        {
+            LogMessage(message, LogLevel.Error, true);
+        }
+
+        /// <summary>
+        /// Method for warning logging
+        /// </summary>
+        public void Warning(string message)
+        {
+            LogMessage(message, LogLevel.Warning, false);
+        }
+
+        /// <summary>
+        /// Method for warning logging of Pii 
+        /// </summary>
+        public void WarningPii(string message)
+        {
+            LogMessage(message, LogLevel.Warning, true);
+        }
+
+        /// <summary>
+        /// Method for information logging
+        /// </summary>
+        public void Info(string message)
+        {
+            LogMessage(message, LogLevel.Info, false);
+        }
+
+        /// <summary>
+        /// Method for information logging for Pii
+        /// </summary>
+        public void InfoPii(string message)
+        {
+            LogMessage(message, LogLevel.Info, true);
+        }
+
+        /// <summary>
+        /// Method for verbose logging
+        /// </summary>
+        public void Verbose(string message)
+        {
+            LogMessage(message, LogLevel.Verbose, false);
+        }
+
+        /// <summary>
+        /// Method for verbose logging for Pii
+        /// </summary>
+        public void VerbosePii(string message)
+        {
+            LogMessage(message, LogLevel.Verbose, true);
+        }
+
+        /// <summary>
+        /// Method for error exception logging
+        /// </summary>
+        public void Error(Exception ex)
+        {
+            Error(ex.ToString());
+        }
+
+        /// <summary>
+        /// Method for error exception logging for Pii
+        /// </summary>
+        public void ErrorPii(Exception ex)
+        {
+            ErrorPii(ex.ToString());
+        }
+
+        #endregion
+
+        private void LogMessage(string logMessage, LogLevel logLevel, bool containsPii)
+        {
+            if (logLevel > ApplicationLogLevel) return;
+
+            //format log message;
+            string correlationId = (CorrelationId.Equals(Guid.Empty))
+                ? "No CorrelationId"
+                : CorrelationId.ToString();
+
+            string log = string.Format(CultureInfo.InvariantCulture, "MSAL {0} {1} {2} [{3} {4}] {5}", MsalIdHelper.GetMsalVersion(),
```

Comment:
>MSAL {0} {1} {2} [{3} {4}] {5} [](start = 70, length = 30)

this format is not compliant with the spec. please confirm that dev spec reflects the same.